### ISSUE TITLE
Rethink components

### ISSUE THREAD

**Issue by larsivsi** [larsivsi/project_peril#34]
I very much dislike the current construct needed to get a component:
```
match node.get_component(ComponentType::TRANSFORM)
{
        Some(comp) =>
        {
                let immutable_comp = comp.borrow();
                let transform_comp = immutable_comp.as_any().downcast_ref::<TransformComponent>().unwrap();
```
It's big and clunky, and cannot be hidden due to borrow/borrow_mut.

Rethink whether there's a better way to achieve the same.

**Comment by larsivsi** [comment]
This will be fixed by cca51cabd55f6f4ba5fcb8f5cd76354729ec1aa6.

Only thing missing is that I want to derive the ComponentType from the generic, rather than passing it:
```
get_component::<TransformComponent>()
//rather than the current
get_component::<TransformComponent>(ComponentType::TRANSFORM)
```

**Comment by larsivsi** [comment]
ComponentType will be fixed by 2918cd87fe82429d0598dfe9c530668017188f98.

**Comment by larsivsi** [comment]
Fixed by the merge above.
